components/brave_wallet_ui/components/extension/sign-panel/sign_in_with_ethereum.tsx

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import Button from '@brave/leo/react/button'
+import { skipToken } from '@reduxjs/toolkit/query/react'
+
+// Types
+import {
+  SignMessageRequest
+} from '../../../constants/types'
+
+// Queries
+import {
+  useAccountQuery
+} from '../../../common/slices/api.slice.extra'
+
+// Components
+import CreateSiteOrigin from '../../shared/create-site-origin/index'
+import {
+  CreateAccountIcon
+} from '../../shared/create-account-icon/create-account-icon'
+
+// Utils
+import {
+  reduceAddress
+} from '../../../utils/reduce-address'
+import { getLocale } from '../../../../common/locale'
+
+// Styles
+import {
+  StyledWrapper,
+  Title,
+  Description,
+  FavIcon,
+  OriginName,
+  MessageBox,
+  MessageText,
+  IconButton,
+  CloseIcon,
+  DetailsKeyText,
+  DetailsInfoText,
+  CodeBlock
+} from './sign_in_with_ethereum.style'
+import {
+  Row,
+  Column,
+  VerticalDivider,
+  VerticalSpace,
+  HorizontalSpace,
+  ScrollableColumn
+} from '../../shared/style'
+
+export const getKeyLocale = (key: string) => {
+  const capitalized = key.charAt(0).toUpperCase() + key.slice(1)
+  const localeString = `braveWallet${capitalized}`
+  return getLocale(localeString) === localeString
+    ? key
+    : getLocale(localeString)
+}
+
+interface Props {
+  data: SignMessageRequest
+  onSignIn: () => void
+  onCancel: () => void
+}
+
+export const SignInWithEthereum = (props: Props) => {
```

Comment:
I think we need to update the UI here to the latest changes that @minkyeongshin finished up. I think we changed them after they got implemented was all.

See https://www.figma.com/file/hOASsXLy3oGhX97XsKR0Ib/Sign-in-with-Brave-Wallet?type=design&node-id=109%3A10271&mode=design&t=2mNtKy03QZOnjsUK-1

Reply:
@kdenhartog I'm on this! Looks like we just need to display `Message` and `Resource` on the first screen if both are available correct?

![Screenshot 73](https://github.com/brave/brave-core/assets/40611140/f72c7842-2454-47a6-b125-6ca184b26e1c)

Reply:
Yup seems like it sense the URI differing case won't appear with the current backend logic. If we add it we can address UI changes later.

Reply:
@kdenhartog I have updated the UI to the latest Design!

https://github.com/brave/brave-core/assets/40611140/8110c05a-1c2e-4306-979c-afb9be650ca4